src/NUnitEngine/nunit.engine/Internal/SettingsStore.cs

```diff
@@ -96,38 +96,46 @@ public void LoadSettings()
 
         public void SaveSettings()
         {
-            if (_writeable)
+            if (_writeable && _settings.Keys.Count > 0)
```

Comment:
No current case. In future we might have a settings editor, but that's not a sure thing. I think we may eventually convert to json and let people edit with their favorite editor.